src/vs/workbench/services/activity/common/activityService.ts

```diff
@@ -70,4 +70,24 @@ export interface IActivityService {
 	 * Clears activity shown in the activitybar for the given viewlet or panel.
 	 */
 	clearActivity(compositeId: string): void;
-}
\ No newline at end of file
+
+	/**
+	 * Get registered external viewlets' info for populating 'Toggle Custom Explorer' command picks.
+	 */
+	getInfoForRegisteredViewlets(): {
+		[viewletId: string]: {
+			isEnabled: boolean;
+			treeLabel: string;
+		}
+	};
+
+	/**
+	 * Enable/disable an external viewlet.
+	 */
+	toggleViewlet(viewletId: string): void;
+
+	/**
+	 * Get the external viewlet id that is about to open.
+	 */
+	getExternalViewletIdToOpen(): string;
+}
```

Comment:
To my understanding `SideBarPart/IViewletService` is more low-level and just does composite construction, whereas `ActivityBarPart/IActivityService` knows all viewlets and which ones are enabled. So I feel the logic should be here instead of in `IViewletService`, but the name `IActivityService` is indeed confusing.